source/agora/consensus/ValidatorSet.d

```diff
@@ -295,9 +295,10 @@ public class ValidatorSet
 
     public void clearExpiredValidators (ulong block_height) @safe nothrow
     {
-        // the smallest enrolled height would be 1, so the passed block height
-        // should be greater than the validator cycle for deleting validators.
-        if (block_height > Enrollment.ValidatorCycle)
+        // the smallest enrolled height would be 0 (genesis block),
+        // so the passed block height should be at minimum the
+        // size of the validator cycle
+        if (block_height >= Enrollment.ValidatorCycle)
```

Comment:
Can we go with an early return here ? It's a bit bigger refactoring but it makes things clearer.

Reply:
Sure, I thought about it too.